docroot/modules/custom/foia_users/foia_users.module

```diff
@@ -40,63 +40,83 @@ function _office_user_nid_access($node, $account) {
             ->cachePerUser()
             ->addCacheableDependency($node);
-        }
-        else {
-          return _office_user_nid_access($node, $account);
-        }
-      }
-      elseif ($op == "update") {
-        return _office_user_nid_access($node, $account);
+
+        default:
+          return AccessResult::neutral();
       }
-      else {
+    }
+  }
+  return AccessResult::neutral();
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_access().
+ */
+function foia_users_taxonomy_term_access(EntityInterface $entity, $operation, AccountInterface $account) {
+  // Get the term ID.
+  $term_id = $entity->id();
+  // Get the current user's roles.
+  $user = User::load(\Drupal::currentUser()->id());
+  $user_roles = $user->getRoles();
+  // Get the current user's agency.
+  $user_agency = $user->get('field_agency')->target_id;
+
+  if ($term_id === $user_agency && in_array('agency_manager', $user_roles)) {
+    switch ($operation) {
+      case 'update':
         return AccessResult::allowed()
           ->cachePerUser()
-          ->addCacheableDependency($node);
-      }
+          ->addCacheableDependency($entity);
 
-    }
+      case 'delete':
+        return AccessResult::forbidden()
+          ->cachePerUser()
+          ->addCacheableDependency($entity);
 
+      default:
+        return AccessResult::neutral();
+    }
   }
   return AccessResult::neutral();
 }
 
 /**
- * Custom submit handler for office value.
+ * Custom submit handler for agency value.
  */
-function _form_validation_foiaoffice($form, $form_state) {
-
-  if ($form_state->hasValue('field_foia_office')) {
-
-    $role_array = ["foia_office_creator", "foia_office_reviewer"];
+function _form_validation_foia_agency($form, $form_state) {
+  if ($form_state->hasValue('field_agency')) {
+    $role_array = [
+      'foia_office_creator',
+      'foia_office_reviewer',
+      'agency_manager',
+    ];
     $user_role = $form_state->getValue('roles');
+    $user_agency = $form_state->getValue('field_agency')[0]['target_id'];
     $user_role_match = FALSE;
-    $field_foia_office = $form_state->getValue('field_foia_office')[0]['target_id'];
-
-    foreach ($user_role as $k => $v) {
-      if (in_array($v, $role_array)) {
-        $user_role_match = TRUE;
-      }
+    if (is_array($user_role)) {
+      $user_role_match = array_intersect($user_role, $role_array) ? TRUE : $user_role_match;
     }
 
-    if (is_null($field_foia_office) && ($user_role_match)) {
-      $form['field_foia_office']['widget']['#required'] = TRUE;
-      $form_state->setErrorByName('field_foia_office', t('The roles "FOIA Office creator" and "FOIA Office reviewer" must be assigned to an office.'));
+    if (is_null($user_agency) && ($user_role_match)) {
+      $form['field_agency']['widget']['#required'] = TRUE;
+      $form_state->setErrorByName('field_agency', t('The roles "Agency Component creator", "Agency Component reviewer" and "Agency Manager" must be assigned to an agency.'));
```

Comment:
We should get rid of mention of `Agency Component creator` and `Agency Component reviewer` from this message since I don't think we'll be using it. Also, maybe we can say 

> Users with the "Agency Manager" role must be assigned to an agency.

Reply:
Agreed. Should I not be checking for those other roles at all? That would simplify the `array_intersect` to an `in_array`.

Reply:
I don't think so - see [comment above](https://github.com/usdoj/foia-api/pull/67#discussion_r141098128)